Docs

johannesjh opened the pull request:
* [x] adds empty lines between bulletpoints
* [x] updates doc-related dependencies
* [x] fixes bullet point rendering #57
* [x] make sure we publish docs whenever a tag is built

fixes #57

johannesjh committed:
adds emtpy lines between bullet points

johannesjh committed:
updates doc dependencies

johannesjh commented:
I added whitespace around bullet points, this did not fix the bulletpoint rendering, at least not on my local machine.

I updated doc-related dependencies.  this is useful on it's own. it did however require to bump the minimum python version to 3.8. the updated dependencies should fix the bulletpoint rendering, judging from what I read over on stackoverflow https://stackoverflow.com/q/67542699, but in this case, it did not fix the issue, at least not on my local machine.

I am pushing the changes to see what the ci-built docs look like. to my surprise, the bulletpoints are now rendered correctly, see https://johannesjh.github.io/req2flatpak/docs/introduction.html

johannesjh commented:
last checkbox, I wanted to review and verify that we actually publish docs for tags that are being build. 

* regarding previous tags: in the [gh-pages branch](https://github.com/johannesjh/req2flatpak/tree/gh-pages), documentation for the tag `v0.1` exists, but versions.txt does not reference it. It may be that I accidentally deleted the reference in versions.txt when I manually cleaned it up a while a go.
  * this means I could try to manually add `v0.1` to versions.txt again

* regarding future tags: the ci script looks good to me: https://github.com/johannesjh/req2flatpak/blob/f898a19ef26036ff11b912085c6451732899bfb9/.github/workflows/ci.yml#L91

real-yfprojects commented:
> I added whitespace around bullet points, this did not fix the bulletpoint rendering, at least not on my local machine.

Same for me however updating dependencies did the trick. Did you `--sync` your poetry environment afterwards?

> regarding future tags: the ci script looks good to me

I saw that you copied it from `sphinx_rtd_theme_github_versions` however it sometimes overrides `version.txt`: https://github.com/johannesjh/req2flatpak/commit/a0100826eeed8c7cb0f8028f94e08c681e5af40b 
I am currently trying to find the reason for that.

real-yfprojects commented:
Ci output:
```
> Run poetry run sphinx_rtd_theme_github_versions .github/pages
fatal: Not a valid object name origin/gh-pages
WARNING:root:Cannot get gh-pages branch contents
Local directory: docs
Tags list: ['']
Sorted versions: ['docs']
```

Maybe running `git fetch origin gh-pages` fixes that.

real-yfprojects committed:
ci: Fetch `gh-pages` branch and tags before generating `versions.txt`.

This should fix an error with the script not being able to find the branch.

real-yfprojects commented:
I ticked the checkbox because I think this is fixed now.

real-yfprojects commented on pyproject.toml:
```diff
@@ -14,7 +14,7 @@ readme      = ["README.rst", "docs/source/changelog.rst"]
 
 [tool.poetry.dependencies]
-python    = "^3.7.2"
+python    = "^3.8"
```
Not sure we want to drop `3.7`. This is a very simple build script that should easily support most python versions. `py3.7` is still used in many distributions or on many installations afaik.

Updating `sphinx-rtd-theme` to `^1.2.0` already fixes the issue. No need to update `sphinx`.

johannesjh replied:
I agree. I'll re-do the dependency updating, keeping compatibility with py3.7

johannesjh replied:
done.

johannesjh committed:
downgrades sphinx to stay compatible with python 3.7.

note, I had to downgrade isort as well because isort >= 5.12 is not compatible with py3.7 anymore.

johannesjh commented:
I downgraded sphinx to stay compatible with py3.7. 
=> I think we are all set, ready for merging.
